service/storage_service.cc

```diff
@@ -917,6 +918,11 @@ class topology_coordinator {
     raft_topology_cmd_handler_type _raft_topology_cmd_handler;
 
     tablet_allocator& _tablet_allocator;
+    seastar::timer<lowres_clock> _load_stats_refresh_trigger;
+    lowres_clock::time_point _last_load_stats_refresh;
+public:
+    static constexpr std::chrono::seconds load_stats_refresh_interval = std::chrono::seconds(5);
```

Comment:
We could try to be clever and make the interval less frequent for tables with many tablets. They can't grow rapidly.

The base interval can be a minute, it's hard to imagine to need faster reaction than that.

Reply:
You're right. 5s is overkill. 

To make frequency per table, we change RPC interface to send a list of table ids for handler, and coordinator keeps a last_update timestamp for each table.
Do you want to reduce the network congestion? The amount of data pulled is nr_of_nodes * nr_of_tables * sizeof(table_load_stats)=16. With nr_of_nodes = 100, nr_of_tables = 5000, it's 8M of data. It seems to me that the most expensive part is pulling the data from each node, and each node aggregating stats from each shard. If we already there, it's not better to put the stats of all tables (which is readily available in replica::table stats)? If table has large tablet count, and avg tablet size is barely moving in that minimum frequency, it means load balancer will only keep ignoring it. A table can be ruled out for resizing decision quickly by looking at avg table size, that is very cheap to calculate.

We also want to make it configurable for testing, as we don't want to inject a sleep for load balancer to take action. Probably piggyback on behavior injection api to avoid adding another config.

Reply:
I just want to reduce work. 

Making it per-table will increase work for clusters with many tables.

Reply:
Even at 40 MB/s/shard, a tablet will grow from 5GB to 10GB in two minutes. That's if there's just one growing tablet on the shard, usually there will be many.

Reply:
Alright makes sense. If a table enter splitting mode, then we can use a exponential backoff retry with base of 1 min (base too low? it depends on table size; so we can estimate the time for each replica to complete splitting work based on total shard count and size), to poll for split completeness.

Reply:
We can also apply the concept of time bucket, such that per-table stat expirations that fall into the bucket can be batched into the same RPC.

Reply:
Let's start with one periodic timer to collect all stats.